Include/Rover/Box.hpp

```diff
@@ -13,57 +13,51 @@ namespace Rover {
     \tparam T The type of value to generate.
   */
   template<typename T>
-  class Box : Noncopyable {
+  class Box : private Noncopyable {
     public:
 
       //! The type of value to generate.
       using Type = T;
 
       //! Packs a generator or generator pointer into a Box
       template<typename GeneratorFwd, std::enable_if_t<!std::is_same_v<
-          std::decay_t<GeneratorFwd>, Box>>* = nullptr>
+        std::decay_t<GeneratorFwd>, Box>>* = nullptr>
       explicit Box(GeneratorFwd&& gen);
 
       Type generate(Evaluator& evaluator);
 
     private:
-
-      struct WrapperBase {
+      class WrapperBase {
+      public:
```

Comment:
```public:``` must be indented and all declarations that follow must be indented another level.